src/python_framework_v02/troute/nhd_network_utilities_v02.py

```diff
@@ -679,3 +679,12 @@ def build_data_assimilation_folder(data_assimilation_parameters):
         )
 
     return usgs_df
+
+def build_reach_type_list(reach_list, wbodies_segs):
+    
+    reach_type_list = [
+                1 if (set(reaches) & wbodies_segs) else 0 for reaches in reach_list
+            ]
+    reaches_list_with_type = list(zip(reach_list, reach_type_list))
```

Comment:
Use `return list(zip(reach_list, reach_type_list))` -- no need to allocate the extra variable.